### PR TITLE
Habitat 2.0 Challenge Tasks

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

Task Changes:
* Added challenge tasks: `rearrange_easy` and `rearrange`.
* Refactored PDDL for complete support of [PDDL 1.2](https://planning.wiki/ref/pddl). The old code was difficult to maintain and had inconsistent support with PDDL. 
* Added Fetch Suction model.
* Added ability to cache robot starts in evaluation sets.
* Added virtual sensor to make the sensor requirements of TP-SRL more clear.

Dataset Generator Changes:
* Refactored receptacle tracking. 
* Added field for maximum placements per receptacle. 

Other Changes:
* Refactored robot and agent data management.
* Updated Habitat 2.0 tutorials.
* Refactored and integrated GFX replay.
* Fixed issue with evaluating TP-SRL method.
* End-to-end test for TP-SRL evaluation.

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat/tasks/rearrange/grip_actions.py [hunk 1]
```diff
@@ -18,7 +19,41 @@
...
+        return self._sim.robots_mgr[self._config.AGENT].grasp_mgr
+
+    @property
+    def _action_arg_prefix(self) -> str:
+        if self._config.AGENT is not None:
+            return f"AGENT_{self._config.AGENT}_"
+        return ""
+
+    @property
+    def _robot_prefix(self) -> str:
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/grip_actions.py:50]: This is unused for now, I think you should remove it.

File: habitat/tasks/rearrange/grip_actions.py [hunk 2]
```diff
@@ -18,7 +19,41 @@
...
+    _sim: RearrangeSim
+
+    @property
+    def _ik_helper(self):
+        if self._config.AGENT is None:
+            return self._sim.robots_mgr[0].ik_helper
+        return self._sim.robots_mgr[self._config.AGENT].ik_helper
+
+    @property
+    def cur_robot(self):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/grip_actions.py:32]: Little docstring

File: habitat/tasks/rearrange/grip_actions.py [hunk 3]
```diff
@@ -18,7 +19,41 @@
 )
 
 
-class GripSimulatorTaskAction(SimulatorTaskAction):
+class RobotAction(SimulatorTaskAction):
+    _sim: RearrangeSim
+
+    @property
+    def _ik_helper(self):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/grip_actions.py:26]: Both _ik_helper and cur_robot use the same logic : 
```
if self._config.AGENT is None:
            return self._sim.robots_mgr[0].XXX
        return self._sim.robots_mgr[self._config.AGENT].XXX
```
I would make a helper method : 
```
def _robot_mgr(self):
   if self._config.AGENT is None:
            return self._sim.robots_mgr[0]
   return self._sim.robots_mgr[self._config.AGENT]
```
And then have _ik_helper and cur_robot just call `self._robot_mgr().ik_helper` or `self._robot_mgr().robot`

File: habitat/tasks/rearrange/marker_info.py [hunk 4]
```diff
@@ -27,7 +27,9 @@ def __init__(self, offset_position, link_node, ao_parent, link_id):
         self.update()
 
     def set_targ_js(self, js):
-        self.ao_parent.joint_positions[self.joint_idx] = js
+        js_arr = self.ao_parent.joint_positions[:]
+        js_arr[self.joint_idx] = js
+        self.ao_parent.joint_positions = js_arr
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/marker_info.py:32]: Why the extra logic?
If the previous code had an issue, I think you should write a unit test to make sure the same bug does not happen again

File: habitat/tasks/rearrange/multi_task/composite_sensors.py [hunk 5]
```diff
@@ -5,14 +5,52 @@
...
+        super().__init__(config=config)
+
+    def _get_uuid(self, *args, **kwargs):
+        return "all_predicates"
+
+    def _get_sensor_type(self, *args, **kwargs):
+        return SensorTypes.TENSOR
+
+    def _get_observation_space(self, *args, config, **kwargs):
+        if self._predicates_list is None:
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/multi_task/composite_sensors.py:38]: Why the lazy initialization of self._predicates_list ?
I think this assumes that get_observation is always called after _get_observation_space and we should not make such assumptions

File: habitat/tasks/rearrange/multi_task/composite_sensors.py [hunk 6]
```diff
@@ -117,62 +155,25 @@ def update_metric(self, *args, episode, task, observations, **kwargs):
 
 
 @registry.register_measure
-class CompositeReward(Measure):
-    """
-    The reward based on where the agent currently is in the hand defined solution list.
-    """
-
-    cls_uuid: str = "composite_reward"
+class CompositeSparseReward(Measure):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/multi_task/composite_sensors.py:158]: Add a comment describing what it does

File: habitat/tasks/rearrange/multi_task/composite_sensors.py [hunk 7]
```diff
@@ -117,62 +155,25 @@ def update_metric(self, *args, episode, task, observations, **kwargs):
...
-        if self._prev_node_idx is None:
-            self._prev_node_idx = node_idx
-        elif node_idx > self._prev_node_idx:
-            self._metric += self._config.STAGE_COMPLETE_REWARD
-
-        if task.forced_node_task is None:
-            cur_task_cfg = task.get_inferrred_node_task()._config
-        else:
-            cur_task_cfg = task.forced_node_task._config
+        self._metric = self._config.SLACK_REWARD
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/multi_task/composite_sensors.py:172]: It looks like it only adds a slack reward, not sure what this reward is or why it is sparse (slack rewards tend to be dense)

**Review by vincentpierre — COMMENTED** [review]
Did a first pass on some of the files, but I will need more time to review everything. 
(Note to self : stopped at habitat/tasks/rearrange/multi_task/composite_task.py)

File: examples/interactive_play.py [hunk 8]
```diff
@@ -70,21 +72,40 @@
...
 
 def get_input_vel_ctlr(
-    skip_pygame, arm_action, g_args, prev_obs, env, not_block_input
+    skip_pygame,
+    arm_action,
+    g_args,
+    prev_obs,
+    env,
+    not_block_input,
+    agent_to_control,
```
[the remark this answers is not in the document]
Comment (ASzot) [reply examples/interactive_play.py:86]: Sure, I will remove this from interactive play.

File: habitat/core/vector_env.py [hunk 9]
```diff
@@ -66,6 +66,7 @@
 CURRENT_EPISODE_NAME = "current_episode"
 NUMBER_OF_EPISODE_NAME = "number_of_episodes"
 ACTION_SPACE_NAME = "action_space"
+ORIG_ACTION_SPACE_NAME = "original_action_space"
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat/core/vector_env.py:69]: I agree this is not ideal, but it is necessary for the TP-SRL method and I imagine also for other methods in the future that handle base and arm actions differently. In the new Gym API, the action space is now a flat vector. This does not tell which parts of the action space correspond to which action. The TP-SRL baseline depends on knowing which part of this vector corresponds to which action. To get that info I need the action dictionary.

File: habitat/datasets/rearrange/combine_datasets.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Script to combine multiple dataset files into one. For example:
+```
+python habitat/datasets/rearrange/combine_datasets.py data/datasets/replica_cad/rearrange/v1/train/rearrange_easy_?.json.gz data/datasets/replica_cad/rearrange/v1/train/rearrange_easy.json.gz
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat/datasets/rearrange/combine_datasets.py:10]: Yes, regex. That is just sample use of the script that I use to run. I will clarify.

File: habitat/tasks/rearrange/multi_task/composite_sensors.py (hunk 5 above)
Comment (ASzot) [reply habitat/tasks/rearrange/multi_task/composite_sensors.py:38]: This is necessary since the task initialization logic has not completed yet. But I removed this assumption and fixed this code.

File: habitat/tasks/rearrange/multi_task/composite_sensors.py (hunk 6 above)
Comment (ASzot) [reply habitat/tasks/rearrange/multi_task/composite_sensors.py:158]: I will just remove in this PR actually.

File: habitat/tasks/rearrange/marker_info.py (hunk 4 above)
Comment (ASzot) [reply habitat/tasks/rearrange/marker_info.py:32]: This was due to some strange bug where the joint position was sometimes not set correctly. I will be challenging to setup an explicit test for this random situation.

File: examples/tutorials/colabs/Habitat2_Quickstart.ipynb [hunk 11]
```diff
@@ -698,7 +651,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!python -m habitat.datasets.rearrange.rearrange_generator --run --config data/nav_pick_dataset.yaml --num-episodes 10 --out data/nav_pick.json.gz"
+    "!python -m habitat.datasets.rearrange.run_rearrange_generator --run --config data/nav_pick_dataset.yaml --num-episodes 10 --out data/nav_pick.json.gz"
```
Comment (vincentpierre) [inline examples/tutorials/colabs/Habitat2_Quickstart.ipynb:654]:
```suggestion
    "!python -m habitat.datasets.rearrange.run_episode_generator --run --config data/nav_pick_dataset.yaml --num-episodes 10 --out data/nav_pick.json.gz"
```

File: habitat/core/benchmark.py [hunk 12]
```diff
@@ -131,6 +133,8 @@ def local_evaluate(
         agg_metrics: Dict = defaultdict(float)
 
         count_episodes = 0
+
+        pbar = tqdm(total=num_episodes)
         while count_episodes < num_episodes:
```
Comment (vincentpierre) [inline habitat/core/benchmark.py:138]: how about 
```
for count_episodes in tqdm(range(num_episodes)):
```
?

File: habitat/tasks/rearrange/grip_actions.py [hunk 13]
```diff
@@ -18,7 +19,61 @@
...
+
+    @property
+    def cur_grasp_mgr(self):
+        """
+        The grasp manager for the robot instance for this action.
+        """
+
+        if self._config.AGENT is None:
+            return self._sim.robots_mgr[0].grasp_mgr
+        return self._sim.robots_mgr[self._config.AGENT].grasp_mgr
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/grip_actions.py:62]:
```suggestion
        return self._robot_mgr.grasp_mgr
```

File: habitat/tasks/rearrange/grip_actions.py [hunk 14]
```diff
@@ -18,7 +19,61 @@
 )
 
 
-class GripSimulatorTaskAction(SimulatorTaskAction):
+class RobotAction(SimulatorTaskAction):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/grip_actions.py:22]: Any reason the `RobotAction` is in `grip_actions.py` instead of `actions.py` ? The separation between what is in `grip_actions.py`vs `actions.py` seems kind of vague.

File: habitat_baselines/utils/common.py [hunk 15]
```diff
@@ -101,18 +101,12 @@ def __init__(
         super().__init__()
 
         self.action_activation = config.action_activation
-        self.use_log_std = config.use_log_std
```
Comment (vincentpierre) [inline habitat_baselines/utils/common.py:104]: If we are removing `use_log_std` from the code, can you remove it from the default config as well ?

File: habitat_baselines/rl/ppo/ppo_trainer.py [hunk 16]
```diff
@@ -982,7 +1000,7 @@ def _eval_checkpoint(
                 logger.warn(f"Evaluating with {total_num_eps} instead.")
                 number_of_eval_episodes = total_num_eps
         assert (
-            number_of_eval_episodes > 1
+            number_of_eval_episodes >= 1
```
Comment (vincentpierre) [inline habitat_baselines/rl/ppo/ppo_trainer.py:1003]:
```suggestion
            number_of_eval_episodes > 0
```

File: habitat_baselines/rl/ppo/ppo_trainer.py [hunk 17]
```diff
@@ -1098,6 +1116,25 @@ def _eval_checkpoint(
...
+                # episode continues
+                elif len(self.config.VIDEO_OPTION) > 0:
+                    # TODO move normalization / channel changing out of the policy and undo it here
+                    frame = observations_to_image(
+                        {k: v[i] for k, v in batch.items()}, infos[i]
+                    )
+                    if self.config.VIDEO_RENDER_ALL_INFO:
+                        frame = overlay_frame(frame, infos[i])
+
+                    rgb_frames[i].append(frame)
```
Comment (vincentpierre) [inline habitat_baselines/rl/ppo/ppo_trainer.py:1136]: This also happens line 1084. Are we doubling up all the frames ?

File: habitat_baselines/rl/hrl/utils.py [hunk 18]
```diff
@@ -0,0 +1,20 @@
+from typing import Tuple
+
+from habitat.core.spaces import ActionSpace
+from habitat_baselines.utils.common import get_num_actions
+
+
+def find_action_range(
+    action_space: ActionSpace, search_key: str
+) -> Tuple[int, int]:
```
Comment (vincentpierre) [inline habitat_baselines/rl/hrl/utils.py:9]: docstring

**Review by vincentpierre — COMMENTED** [review]
I really wish I were able to review +6,412 −4,683 PR but I cannot. I am not even done reviewing 20% of it. Can I ask you to make sure you put docstrings on all public methods, I think I will end up accepting the change without understanding it...

File: habitat/tasks/rearrange/sub_tasks/articulated_object_sensors.py [hunk 19]
```diff
@@ -50,7 +50,7 @@ def _get_observation_space(self, *args, **kwargs):
 
     def get_observation(self, observations, episode, *args, **kwargs):
         marker = self._task.get_use_marker()
-        ee_trans = self._sim.robot.ee_transform
+        ee_trans = self._sim.get_robot_data(self.robot_id).robot.ee_transform
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/sub_tasks/articulated_object_sensors.py:53]: won't `self.robot_id` always be none and thus get_robot_data always return `robots_mgr[0]` ?

File: habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py [hunk 20]
```diff
@@ -131,9 +140,15 @@ def get_observation(self, task, *args, **kwargs):
 
 
 @registry.register_sensor
-class NavGoalSensor(Sensor):
+class NavGoalSensor(UsesRobotInterface, Sensor):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py:143]: Why does this use the `UsesRobotInterface` but does not use self.robot_id? Seems like this inheritance is not useful

File: habitat/tasks/rearrange/utils.py [hunk 21]
```diff
@@ -352,6 +349,27 @@ def calc_ik(self, targ_ee: np.ndarray):
...
+            self.pb_link_idx,
+            targ_ee,
+            physicsClientId=self.pc_id,
         )
         return js[: self._arm_len]
+
+
+class UsesRobotInterface:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/utils.py:362]: I doubt this super().__init__ call is useful.
This whole class does not seem very useful (at least for now), can you add some documentation or a TODO so we know what to do with it in the future?

File: habitat/tasks/rearrange/utils.py [hunk 22]
```diff
@@ -352,6 +349,27 @@ def calc_ik(self, targ_ee: np.ndarray):
...
+        self.robot_id = None
+
+
+def write_gfx_replay(gfx_keyframe_str, task_config, ep_id):
```
Comment (vincentpierre) [inline habitat/tasks/rearrange/utils.py:366]: docstring

File: habitat_baselines/common/base_trainer.py [hunk 23]
```diff
@@ -98,11 +98,19 @@ def eval(self) -> None:
             ), "Must specify a directory for storing videos on disk"
 
         with get_writer(self.config, flush_secs=self.flush_secs) as writer:
-            if os.path.isfile(self.config.EVAL_CKPT_PATH_DIR):
+            if (
+                os.path.isfile(self.config.EVAL_CKPT_PATH_DIR)
+                or not self.config.EVAL.SHOULD_LOAD_CKPT
```
Comment (vincentpierre) [inline habitat_baselines/common/base_trainer.py:103]: Why does SHOULD_LOAD_CKPT influence wether or not we should evaluate a single checkpoint? Is I have no idea what SHOULD_LOAD_CKPT does at all but it looks like if 
```
SHOULD_LOAD_CKPT = False
EVAL_CKPT_PATH_DIR = directory
```
Then you will call _eval_checkpoint with a directory which will raise an error.

Why is this change needed ?

File: habitat_baselines/common/obs_transformers.py [hunk 24]
```diff
@@ -1228,3 +1228,37 @@ def apply_obs_transforms_obs_space(
     for obs_transform in obs_transforms:
         obs_space = obs_transform.transform_observation_space(obs_space)
     return obs_space
+
+
+@baseline_registry.register_obs_transformer()
+class AddVirtualKeys(ObservationTransformer):
```
Comment (vincentpierre) [inline habitat_baselines/common/obs_transformers.py:1234]: docstrings

File: habitat_baselines/utils/common.py (hunk 15 above)
Comment (vincentpierre) [reply habitat_baselines/utils/common.py:104]: nevermind, I see you removed it.

File: habitat_baselines/rl/hrl/skills/nav.py [hunk 25]
```diff
@@ -77,7 +77,7 @@ def _is_skill_done(
         return should_stop
 
     def _parse_skill_arg(self, skill_arg):
-        targ_name, targ_idx = skill_arg[-1].split("|")
+        targ_name, targ_idx = skill_arg[-2].split("|")
```
Comment (vincentpierre) [inline habitat_baselines/rl/hrl/skills/nav.py:80]: Was this a bug or did the code change in such a way that the argument position changed. 
In either case, can you add a unit test?

File: habitat/core/benchmark.py (hunk 12 above)
Comment (ASzot) [reply habitat/core/benchmark.py:138]: I agree this is better, but this code touches all the other challenges as well, so I am worried about consequences of this decision that I don't understand.

File: habitat/tasks/rearrange/grip_actions.py (hunk 14 above)
Comment (ASzot) [reply habitat/tasks/rearrange/grip_actions.py:22]: `RobotAction` belongs in neither. Created a new file for this. `grip_actions` should be grip only actions, `actions.py` is everything else.

File: habitat_baselines/rl/ppo/ppo_trainer.py (hunk 17 above)
Comment (ASzot) [reply habitat_baselines/rl/ppo/ppo_trainer.py:1136]: Good catch, this was an issue with one of my merges.

File: habitat/tasks/rearrange/multi_task/pddl_action.py [hunk 26]
```diff
@@ -0,0 +1,187 @@
...
+@dataclass
+class ActionTaskInfo:
+    task_config: Optional[Config]
+    task: str
+    task_def: str
+    config_args: Dict[str, Any] = field(default_factory=dict)
+    add_task_args: Dict[str, PddlEntity] = field(default_factory=dict)
+
+
+class PddlAction:
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat/tasks/rearrange/multi_task/pddl_action.py:35]: I am referring to the PDDL terminology with `parameters`, but I will add docstring.

File: habitat/tasks/rearrange/multi_task/pddl_set_state.py [hunk 27]
```diff
@@ -0,0 +1,354 @@
...
+
+            if entity.expr_type.name == target.expr_type.name:
+                return False
+
+        return all(
+            type_matches(art_entity, ART_OBJ_TYPE)
+            for art_entity in self._art_states
+        )
+
+    def is_true(
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat/tasks/rearrange/multi_task/pddl_set_state.py:236]: No, I think `is_true` is more suitable since it checks if the conditions are true in the simulator state.

File: habitat/tasks/rearrange/sub_tasks/articulated_object_sensors.py (hunk 19 above)
Comment (ASzot) [reply habitat/tasks/rearrange/sub_tasks/articulated_object_sensors.py:53]: Yes... this is true, but it is needed for future PR of multi-agent support. Robot manager was so entangled in all this code that it was hard for me to separate it out.

File: habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py (hunk 20 above)
Comment (ASzot) [reply habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py:143]: Good catch. Removed.

File: habitat_baselines/common/base_trainer.py (hunk 23 above)
Comment (ASzot) [reply habitat_baselines/common/base_trainer.py:103]: I think it is best this way actually. When `SHOULD_LOAD_CKPT = False` only a single eval should run no matter what since `EVAL_CKPT_PATH_DIR`. So the situation you described is okay since `EVAL_CKPT_PATH_DIR` is ignored. 

This change is necessary because for models that do not load from a single checkpoint we should not iterate through the default `EVAL_CKPT_PATH_DIR` argument (which is a directory by default).

Comment (ASzot) [reply habitat_baselines/common/base_trainer.py:103]: I added a comment about this.

File: habitat_baselines/rl/hrl/skills/nav.py (hunk 25 above)
Comment (ASzot) [reply habitat_baselines/rl/hrl/skills/nav.py:80]: Not a bug before, I changed the code so that this argument was changed. This is already tested in the composite task tests.

**Review by vincentpierre — APPROVED** [review]
If possible, I would like a second reviewer to approve since this is a large change.

File: .circleci/config.yml [hunk 28]
```diff
@@ -266,7 +266,7 @@ jobs:
               export PYTHONPATH=.:$PYTHONPATH
               # This is a flag that enables test_test_baseline_training to work
               export TEST_BASELINE_SMALL=1
-              python setup.py test --addopts -v -s "test/test_baseline_training.py"
+              python setup.py test --addopts "test/test_baseline_training.py"
```
Comment (vincentpierre) [inline .circleci/config.yml:269]: python setup.py test --addopts "-s  test/test_baseline_training.py"   #`-s` is for verbose

**Review by mathfac — APPROVED** [review]
Thank you, looks like great improvement from previous version. Let's keep further refactoring and development in separate PR.

File: configs/tasks/rearrange/benchmark/idle_single_camera.yaml [hunk 29]
```diff
@@ -51,16 +51,15 @@ SIMULATOR:
         SENSORS: ['HEAD_RGB_SENSOR']
         START_POSITION: [3.66997,0.0,3.5041]
         START_ROTATION: [0.0, 0.999657, 0.0, -0.0261769]
+        ROBOT_TYPE: FetchRobotNoWheels
+        ROBOT_URDF: ./data/robots/hab_fetch/robots/fetch_no_base.urdf
     HEAD_RGB_SENSOR:
         WIDTH: 128
         HEIGHT: 128
 
     # Agent setup
```
Comment (mathfac) [inline configs/tasks/rearrange/benchmark/idle_single_camera.yaml:60]: One of todo: move all these options related to the agent into AGENT's node.

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 30]
```diff
@@ -52,7 +52,7 @@ def __init__(
         policy_config: Config = None,
         aux_loss_config: Optional[Config] = None,
         fuse_keys: Optional[List[str]] = None,
-        **kwargs
+        **kwargs,
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:55]: Not sure what was rational to remove it. Idea adding keyword arguments is that for policy initialization through the constructor we will have different set of parameters for different type of policies. These way we can still use same repository of policies, but have different APIs for policies handling different tasks.

File: habitat_baselines/rl/hrl/hierarchical_policy.py [hunk 31]
```diff
@@ -53,7 +55,11 @@ def __init__(
 
             cls = eval(skill_config.skill_name)
             skill_policy = cls.from_config(
-                skill_config, observation_space, action_space, self._num_envs
+                skill_config,
+                observation_space,
+                action_space,
+                self._num_envs,
+                full_config,
```
Comment (mathfac) [inline habitat_baselines/rl/hrl/hierarchical_policy.py:62]: config, full_config, skill_config is a little bit confusing. We can name it with main node name: like policy_config. Overall if Policy need access to whole main config then it is better to pass specifics parts that needed from that main config. Otherwise we have uncontrolled access to config parameters across modules.

File: habitat_baselines/rl/hrl/high_level_policy.py [hunk 32]
```diff
@@ -19,11 +19,22 @@ def get_next_skill(
 
 
 class GtHighLevelPolicy:
```
Comment (mathfac) [inline habitat_baselines/rl/hrl/high_level_policy.py:21]: Doc for class needed.

File: habitat_baselines/rl/hrl/skills/nav.py (hunk 25 above)
Comment (mathfac) [inline habitat_baselines/rl/hrl/skills/nav.py:80]: That would be great to add doc to the class, what type of navigation it can handle. Curious if we can avoid parsing and splitting strings for arguments communication across components and use some named structures instead.

File: habitat_baselines/rl/hrl/skills/nn_skill.py [hunk 33]
```diff
@@ -113,34 +115,31 @@ def _internal_act(
         return full_action, rnn_hidden_states
 
     @classmethod
-    def from_config(cls, config, observation_space, action_space, batch_size):
+    def from_config(
+        cls, config, observation_space, action_space, batch_size, full_config
```
Comment (mathfac) [inline habitat_baselines/rl/hrl/skills/nn_skill.py:119]: Same here regarding full config. What part of it is used and what is the difference between config and full_config.

**Review by mathfac — COMMENTED** [review]
First set of comments we can act after merge.
